backend/functions/lib/adapters/types/attorney.class.ts

```diff
@@ -1,56 +1,52 @@
-export class Attorney {
-  private _firstName: string;
-  private _middleName: string;
-  private _lastName: string;
-  private _generation: string;
-  private _courtId: string;
-  private _office: string;
-
-  get firstName(): string {
-    return this._firstName;
-  }
-
-  set firstName(value: string) {
-    this._firstName = value;
-  }
-
-  get middleName(): string {
-    return this._middleName;
-  }
-
-  set middleName(value: string) {
-    this._middleName = value;
-  }
-
-  get lastName(): string {
-    return this._lastName;
-  }
-
-  set lastName(value: string) {
-    this._lastName = value;
-  }
-
-  get generation(): string {
-    return this._generation;
-  }
-
-  set generation(value: string) {
-    this._generation = value;
-  }
-
-  get courtId(): string {
-    return this._courtId;
-  }
-
-  set courtId(value: string) {
-    this._courtId = value;
-  }
+import { ObjectKeyVal } from './basic';
 
-  get office(): string {
-    return this._office;
-  }
-
-  set office(value: string) {
-    this._office = value;
+export class Attorney {
+  firstName: string;
+  middleName: string;
+  lastName: string;
+  generation: string;
+  courtId: string;
+  office: string;
+  caseLoad: number;
+
+  constructor(attorney?: ObjectKeyVal) {
+    if (attorney) {
+      Object.entries(attorney).forEach(([key, value]) => {
+        if (key === 'firstName') {
+          this.firstName = value as string;
+        } else if (key === 'middleName') {
+          this.middleName = value as string;
+        } else if (key === 'lastName') {
+          this.lastName = value as string;
+        } else if (key === 'generation') {
+          this.generation = value as string;
+        }
+      });
+    }
+  }
+
+  getFullName(fullMiddle = false) {
+    let full = this.firstName;
+    if (this.middleName) {
+      const middle = fullMiddle ? ' ' + this.middleName : ' ' + this.middleName.slice(0, 1);
+      full += middle;
+    }
+    full += ' ' + this.lastName;
+    if (this.generation) {
+      full += ' ' + this.generation;
+    }
+    return full;
+  }
+
+  getAsObjectKeyVal(): ObjectKeyVal {
+    return {
+      firstName: this.firstName,
+      middleName: this.middleName,
+      lastName: this.lastName,
+      generation: this.generation,
+      courtId: this.courtId,
```

Comment:
Why is the courtId here? The trial attorneys are not tied to a court, they are tied to an USTP office.

Reply:
I thought we had already removed that.

Reply:
Good catch. It is now removed.